### DISCUSSION TITLE
edgex-ui-go

### DISCUSSION THREAD

**Discussion by MimiXav** [edgexfoundry/community#309] [Q&A]
Hi Edgex's team,

I am working with EdgeX (docker images) (I used latest images 3.1.1, but not for edgex-ui-go(latest 3.1.0)

My question is related on deviceCommands on edgex UI, and on c-sdk

First question, issue

Here a snippet of my json profile file:
....
"deviceCommands":
  [
    {
      "name": "test",
      "readWrite": "RW",
      "isHidden":false,
      "resourceOperations":
      [
        { "deviceResource": "id" },
        { "deviceResource": "param1" }
      ]
    }
  ],
 "deviceResources":
  [
      {
         "name": "id",
         "isHidden":false,
         "attributes": { "parameter": "id"},
          "properties": { "valueType": "String", "readWrite": "RW"}
    },
    {
      "name": "param1",
      "isHidden":true,
      "attributes": { "parameter": "param1" },
      "properties": { "valueType": "Uint8", "readWrite": "RW"}
    }
  ]
}

If I put the value type other than string (i.e. Uint8) The edgex-ui-go (try button) send the request to the edgex-code-command with a json string,  instead of int, so the edgex-core-command failed to parse the data (Unable to parse .... for param1

Second question with the c-sdk, on the gethandler callback, how to retrieve the parameter for a get? with a put and this signature (devsdk_handle_put) we have *values[], so it's not a problem, but for the get method (devsdk_handle_get typedef) I don't know how to retrieve all get parameters

Any suggestion, helps, will be appreciate

Thanks a lot

Regards

**Comment by jiekechoo** [comment]
I can't parse your json profile, please upload it again.

And, you should locate the code line number by the sdk or device service.

**Comment by cloudxxx8** [comment]
For the first question, do you encounter issues if you use edgex-core-command REST API directly?
The current edgex-ui is not tested thourough.  If you confirm it's bug, you can open an issue to the edgex-ui-go repository.  We can see if there is anyone from the community can fix it.

For the second question, GET API doesn't support the payload, so there is not parameter to retrieve.

**Reply by MimiXav** [reply]
Hi,

Perfect, thanks a lot, I will open an issue for edgex-ui, for the first question, with only device name, like parameter, actually it's enough for me at this time.

Thanks again
